src/nvim/drawscreen.c

```diff
@@ -696,13 +734,53 @@ static void win_redr_border(win_T *wp)
   int *adj = wp->w_border_adj;
   int irow = wp->w_height_inner + wp->w_winbar_height, icol = wp->w_width_inner;
 
+  bool title = wp->w_float_config.title;
+
   if (adj[0]) {
     grid_puts_line_start(grid, 0);
     if (adj[3]) {
       grid_put_schar(grid, 0, 0, chars[0], attrs[0]);
     }
-    for (int i = 0; i < icol; i++) {
-      grid_put_schar(grid, 0, i + adj[3], chars[1], attrs[1]);
+
+    if (title) {
+      int len, title_col = 0;
+      VirtText title_texts = wp->w_float_config.title_chunks;
+
+      if (wp->w_float_config.title_text != NULL) {
```

Comment:
can't use len here, needs width of the string. save the pointer return value of `parse_virt_text` in the config struct and use it.